<launch>
  <include file="$(find wheelchair)/launch/kinect2_fovis.launch"/>
  <!--Sensor filter-->
  <param name="/odom" value="odom"/>
  <param name="/base_frame" value="base_footprint"/>
  <node pkg="wheelchair" type="navigation_filter.py" name="navigation_filter"/>

  <!--Fundamental TF broadcaster-->
  <node pkg="wheelchair" type="convert_kinect_frame.py" name="tf_broadcaster"/> 
    
  <!--Trajectory server-->
  <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="hector_trajectory_server">
    <param name="target_frame_name" value="odom"/>
    <param name="source_frame_name" value="base_link"/>
  </node>

  <!-- AMCL -->
  <arg name="custom_amcl_launch_file" default="$(find wheelchair)/launch/includes/kinect_amcl.launch.xml"/>
  <arg name="initial_pose_x" default="0.0"/> <!-- Use 17.0 for willow's map in simulation -->
  <arg name="initial_pose_y" default="0.0"/> <!-- Use 17.0 for willow's map in simulation -->
  <arg name="initial_pose_a" default="0.0"/>
  <include file="$(arg custom_amcl_launch_file)">
    <arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
    <arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
    <arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
  </include>   
  <!-- Move base -->
  <include file="$(find wheelchair)/launch/includes/move_base.launch.xml"/>
</launch>